src/Mailchimp/Synchronisation/RequestBuilder.php

```diff
@@ -332,7 +334,16 @@ private function setZone(Zone $zone): void
             return;
         }
 
-        $fieldName = 'zone'.ucfirst($zone->getType());
+        switch ($zone->getType()) {
+            case Zone::FOREIGN_DISTRICT:
+                $fieldName = 'zoneForeignDistrict';
+
+                break;
+            default:
+                $fieldName = 'zone'.ucfirst($zone->getType());
+
+                break;
+        }
```

Comment:
could you try new String component?
```suggestion
        $fieldName = 'zone'.ucfirst((new ByteString($zone->getType()))->camel());
```